### ISSUE TITLE
[Distributions] Why exp of log_prob produces values larger than 1 ?

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#7637]
Suppose we have following code

```
import torch

torch.manual_seed(1)

d = Normal(loc=torch.randn(10), scale=torch.rand(10))
d.log_prob(d.sample()).exp().max()
```
It gives `tensor(13.7150)`

**Comment by apaszke** [comment]
@fritzo

**Comment by fritzo** [comment]
`Normal` is a continuous probability distribution, so its `.log_prob()` method returns the log of a probability density, not a probability mass function. Of course probability density is not bounded above. E.g. the maximum density of a normal distribution with standard deviation `sigma`
```py
density = Normal(0,sigma).log_prob(0).exp()
```
is equal to `1/sqrt(2 * pi * sigma**2)`, so the maximum density will be greater than 1 whenever `sigma` is narrower than `1/sqrt(2*pi)`.
```py
>>> import math
>>> from torch.distributions import Normal
>>> Normal(0, 1.0 * (2*math.pi)**-0.5).log_prob(0).exp()
tensor(1.)
>>> Normal(0, 2.0 * (2*math.pi)**-0.5).log_prob(0).exp()
tensor(0.5000)
>>> Normal(0, 0.5 * (2*math.pi)**-0.5).log_prob(0).exp()
tensor(2.)
```

**Comment by richardrl** [comment]
> `Normal` is a continuous probability distribution, so its `.log_prob()` method returns the log of a probability density, not a probability mass function. Of course probability density is not bounded above. E.g. the maximum density of a normal distribution with standard deviation `sigma`
> 
> ```python
> density = Normal(0,sigma).log_prob(0).exp()
> ```
> 
> is equal to `1/sqrt(2 * pi * sigma**2)`, so the maximum density will be greater than 1 whenever `sigma` is narrower than `1/sqrt(2*pi)`.
> 
> ```python
> >>> import math
> >>> from torch.distributions import Normal
> >>> Normal(0, 1.0 * (2*math.pi)**-0.5).log_prob(0).exp()
> tensor(1.)
> >>> Normal(0, 2.0 * (2*math.pi)**-0.5).log_prob(0).exp()
> tensor(0.5000)
> >>> Normal(0, 0.5 * (2*math.pi)**-0.5).log_prob(0).exp()
> tensor(2.)
> ```

Shouldn't you rename this log_pdf? Probability is usually used to describe the normalized function
